src/RelayClient.ts

```diff
@@ -232,68 +226,76 @@ class RelayClient extends EnvelopingEventEmitter {
       to,
       tokenAmount,
       tokenContract,
-      tokenGas,
+      tokenGas: constants.Zero,
       value,
     };
 
     const completeRequest: EnvelopingRequest = isDeployment
       ? {
-          request: {
-            ...commonRequestBody,
-            ...{
-              index,
-              recoverer,
-            },
+        request: {
+          ...commonRequestBody,
+          ...{
+            index,
+            recoverer,
           },
-          relayData,
-        }
+        },
+        relayData,
+      }
       : {
-          request: {
-            ...commonRequestBody,
-            ...{
-              gas: BigNumber.from(gasLimit),
-            },
+        request: {
+          ...commonRequestBody,
+          ...{
+            gas: BigNumber.from(gasLimit),
           },
-          relayData,
-        };
+        },
+        relayData,
+      };
 
     return completeRequest;
   };
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore FIXME: remove the ts ignore once the method is consumed
-  private _prepareHttpRequest = async (
-    { feesReceiver: feesReceiverFromHubInfo }: HubInfo,
+  private async _prepareHttpRequest(
+    { feesReceiver }: HubInfo,
     relayRequest: EnvelopingRequest
-  ): Promise<EnvelopingTxRequest> => {
+  ): Promise<EnvelopingTxRequest> {
     const {
-      request: { relayHub },
-      relayData: { callForwarder, feesReceiver: feesReceiverFromRequest },
+      request: { relayHub, tokenContract, tokenAmount, tokenGas },
+      relayData: { callForwarder, gasPrice },
     } = relayRequest;
     const chainId = (await this._provider.getNetwork()).chainId;
     const accountManager = new AccountManager(this._provider, chainId);
     const relayMaxNonce =
       (await this._provider.getTransactionCount(callForwarder)) +
       this._envelopingConfig.maxRelayNonceGap;
 
-    const feesReceiver =
-      feesReceiverFromRequest &&
-      (await feesReceiverFromRequest) !== constants.AddressZero
-        ? feesReceiverFromRequest
-        : feesReceiverFromHubInfo;
     if (
       !feesReceiver ||
       feesReceiver === constants.AddressZero ||
-      !isAddress(await feesReceiver)
+      !isAddress(feesReceiver)
     ) {
       throw new Error('FeesReceiver has to be a valid non-zero address');
     }
 
+    const currentTokenGas = BigNumber.from(tokenGas);
+
+    const newTokenGas = currentTokenGas.gt(constants.Zero) ?
```

Comment:
I think that we could add a forceTokenGas in the requestConfig

Reply:
Yes, indeed. Without it, we cannot reproduce the exact same functionality we had before.

Reply:
I updated, I added an extra param but Im not pretty sure about it, Im also thinking creating a new method for this, but I would like to read more opinions.

Reply:
Got no opinion on the `forceTokenGas` as I don't know the interactions between individual parts, but the `currentTokenGas.gt(constants.Zero)` can be written as `!currentTokenGas.isZero()` ;)

Reply:
updated but this is open to discussion

Reply:
Sorry, my fault. At the beginning, I understood you were referring to `forceGas`. Indeed, we had to apply the logic about it, but now that I checked again, I'm sure we don't need `forceTokenGas` because that field will never be overwritten by the provider being a custom property.